src/csexp_rpc/csexp_rpc.ml

```diff
@@ -56,10 +56,7 @@ module Session = struct
     | Open { in_channel; out_channel; reader; writer; socket } ->
       Worker.stop reader;
       Worker.stop writer;
-      if socket then
-        Unix.shutdown (Unix.descr_of_out_channel out_channel) Unix.SHUTDOWN_ALL
-      else
-        close_in_noerr in_channel;
+      if not socket then close_in_noerr in_channel;
```

Comment:
What a gem. :-)
It would benefit from a comment because it will probably look super-confusing to uninitiated.
The behavior seems right though.

Reply:
I should really just get rid of `dune rpc init` and let clients establish the connections directly. We wouldn't need the `socket = false` case anymore then.

Reply:
I mean, false or not false, if you're only closing one of the two channels, that probably deserves an explanation.

Reply:
(I can see you pushed one)